web/tests/filter.test.js

```diff
@@ -397,6 +397,25 @@ test("basics", () => {
     assert.ok(filter.is_conversation_view());
     assert.ok(!filter.is_conversation_view_with_near());
 
+    terms = [
+        {operator: "channel", operand: "foo"},
+        {operator: "topic", operand: "bar"},
+        {operator: "with", operand: 17},
+    ];
+    filter = new Filter(terms);
+
+    assert.ok(!filter.is_keyword_search());
+    assert.ok(filter.can_mark_messages_read());
+    assert.ok(filter.supports_collapsing_recipients());
+    assert.ok(!filter.contains_only_private_messages());
+    assert.ok(filter.allow_use_first_unread_when_narrowing());
+    assert.ok(filter.includes_full_stream_history());
+    assert.ok(filter.can_apply_locally());
+    assert.ok(!filter.is_personal_filter());
+    assert.ok(!filter.is_conversation_view());
+    assert.ok(filter.can_bucket_by("channel", "topic", "with"));
+    assert.ok(!filter.is_conversation_view_with_near());
```

Comment:
So, I'm not sure whether `filter.is_conversation_view` should be true for these, or `filter.is_conversation_view_with_near`, but it seems like if neither is true, we need a `filter.is_conversation_view_with_with` and to audit all places we check one of the above for this.

I guess a possible theory is that the outcome of a `with` operator should be that you end up with a normal conversation view `filter` with the `with` operator removed. But I don't think that's what you have implemented.

Reply:
Opened #30863 to track what is likely multiple bugs related to this detail.